File: 3871-count-commas-in-range-ii/3871-count-commas-in-range-ii.java
class Solution {
    public long countCommas(long n) {
        long k = 0;

        long x = n;
        while (x >= 1000) {
            x /= 1000;
            k++;
        }

        return (long)(k * (n + 1) - (Math.pow(1000, k + 1) - 1000) / 999);
    }
}

File: 3871-count-commas-in-range-ii/3871-count-commas-in-range-ii.cpp
class Solution {
public:
    long long countCommas(long long n) {
        long long k = 0;

        long long x = n;
        while (x >= 1000) {
            x /= 1000;
            k++;
        }

        return k * (n + 1) - (pow(1000, k+1) - 1000) / 999;
    }
};